crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -50,7 +52,13 @@ impl Deref for BindGroup {
 
     #[inline]
     fn deref(&self) -> &Self::Target {
-        &self.value
+        render_resource_ref!(self.value, wgpu::BindGroup)
+    }
+}
+
+impl Drop for BindGroup {
+    fn drop(&mut self) {
+        render_resource_drop!(&mut self.value, wgpu::BindGroup);
```

Comment:
This doesn't quite work. This will cause the Arc to be dropped twice. You need to wrap the field in ManuallyDrop in the struct definition to prevent it from being dropped after this drpp impl runs.

Reply:
i thought it caused the original arc (which is taken and transmuted) to be dropped once, and leaves a new `Arc<Box()>>` (created by std::mem::taking) in the struct that can be dropped without issues, is that wrong?

Reply:
Missed the mem::take.